src/node_quic_crypto.h

```diff
@@ -1261,549 +313,63 @@ inline int Server_Transport_Params_Parse_CB(
-        map->emplace("ip", value.substr(11));
-      } else if (value.compare(0, 4, "URI:") == 0) {
-        url::URL url(value.substr(4));
-        if (url.flags() & url::URL_FLAGS_CANNOT_BE_BASE ||
-            url.flags() & url::URL_FLAGS_FAILED) {
-          continue;  // Skip this one
-        }
-        map->emplace("uri", url.host());
-      }
-    }
-  }
-  sk_GENERAL_NAME_pop_free(names, GENERAL_NAME_free);
-  bio.reset();
-}
-
-inline bool SplitHostname(
-    const char* hostname,
-    std::vector<std::string>* parts,
-    const char delim = '.') {
-  static std::string check_str =
-      "\x21\x22\x23\x24\x25\x26\x27\x28\x29\x2A\x2B\x2C\x2D\x2E\x2F\x30"
-      "\x31\x32\x33\x34\x35\x36\x37\x38\x39\x3A\x3B\x3C\x3D\x3E\x3F\x40"
-      "\x41\x42\x43\x44\x45\x46\x47\x48\x49\x4A\x4B\x4C\x4D\x4E\x4F\x50"
-      "\x51\x52\x53\x54\x55\x56\x57\x58\x59\x5A\x5B\x5C\x5D\x5E\x5F\x60"
-      "\x61\x62\x63\x64\x65\x66\x67\x68\x69\x6A\x6B\x6C\x6D\x6E\x6F\x70"
-      "\x71\x72\x73\x74\x75\x76\x77\x78\x79\x7A\x7B\x7C\x7D\x7E\x7F";
-
-  std::stringstream str(hostname);
-  std::string part;
-  while (getline(str, part, delim)) {
-    // if (part.length() == 0 ||
-    //     part.find_first_not_of(check_str) != std::string::npos) {
-    //   return false;
-    // }
-    for (size_t n = 0; n < part.length(); n++) {
-      if (part[n] >= 'A' && part[n] <= 'Z')
-        part[n] = (part[n] | 0x20);  // Lower case the letter
-      if (check_str.find(part[n]) == std::string::npos)
-        return false;
-    }
-    parts->push_back(part);
-  }
-  return true;
-}
-
-
-inline bool CheckCertNames(
-    const std::vector<std::string>& host_parts,
-    const std::string& name,
-    bool use_wildcard = true) {
-
-  if (name.length() == 0)
-    return false;
-
-  std::vector<std::string> name_parts;
-  if (!SplitHostname(name.c_str(), &name_parts))
-    return false;
-
-  if (name_parts.size() != host_parts.size())
-    return false;
-
-  for (size_t n = host_parts.size() - 1; n > 0; --n) {
-    if (host_parts[n] != name_parts[n])
-      return false;
-  }
-
-  if (name_parts[0].find("*") == std::string::npos ||
-      name_parts[0].find("xn--") != std::string::npos) {
-    return host_parts[0] == name_parts[0];
-  }
-
-  if (!use_wildcard)
-    return false;
-
-  std::vector<std::string> sub_parts;
-  SplitHostname(name_parts[0].c_str(), &sub_parts, '*');
+    uint64_t verification_expiration);
 
-  if (sub_parts.size() > 2)
-    return false;
+int VerifyPeerCertificate(SSL* ssl);
 
-  if (name_parts.size() <= 2)
-    return false;
+std::string GetCertificateCN(X509* cert);
 
-  std::string prefix;
-  std::string suffix;
-  if (sub_parts.size() == 2) {
-    prefix = sub_parts[0];
-    suffix = sub_parts[1];
-  } else {
-    prefix = "";
-    suffix = sub_parts[0];
-  }
-
-  if (prefix.length() + suffix.length() > host_parts[0].length())
-    return false;
+void GetCertificateAltNames(
+    X509* cert,
+    std::unordered_multimap<std::string, std::string>* map);
```

Comment:
This could just be a return value here, right?

Reply:
Yeah, it could.